src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -125,7 +125,9 @@ void ClusterTaskManager::DispatchScheduledTasksToWorkers(
       worker->SetAllocatedInstances(allocated_instances);
     }
     worker->AssignTaskId(spec.TaskId());
-    worker->AssignJobId(spec.JobId());
+    if (!RayConfig::instance().enable_multi_tenancy()) {
```

Comment:
Why do we need this if? It looks like we already check this inside `AssignJobId` method.

Reply:
Sure. I'll revert this. The check inside `AssignJobId` is for safety concern. Actually, the `AssignJobId` will be removed once we can pass the job ID to worker processes via an env variable, in a later PR.